src/components/exercises/cards.cjsx

```diff
@@ -13,6 +13,39 @@ ScrollTo = require '../scroll-to-mixin'
 ChapterSection = require '../task-plan/chapter-section'
 Icon = require '../icon'
 
+PreviewWrapper = React.createClass
+
+  propTypes:
+    exercise:               React.PropTypes.object.isRequired
+    onShowDetailsViewClick: React.PropTypes.func.isRequired
+    onExerciseToggle:       React.PropTypes.func.isRequired
+    getExerciseIsSelected:  React.PropTypes.func.isRequired
+    getExerciseActions:     React.PropTypes.func.isRequired
+    watchStore:             React.PropTypes.func
+    watchEvent:             React.PropTypes.string
+
+  componentWillMount: ->
+    { watchStore, watchEvent, exercise } = @props
+    watchStore?.on("#{watchEvent}#{exercise.id}", @update)
+  componentWillUnmount: ->
+    { watchStore, watchEvent, exercise } = @props
+    watchStore?.off("#{watchEvent}#{exercise.id}", @update)
+
+  update: -> @setState({})
+
+  render: ->
+    { exercise } = @props
+    <ExercisePreview
+      key={exercise.id}
+      className='exercise-card'
+      isInteractive={false}
+      isVerticallyTruncated={true}
+      isSelected={@props.getExerciseIsSelected(exercise)}
+      exercise={exercise}
+      onOverlayClick={@onExerciseToggle}
+      overlayActions={@props.getExerciseActions(exercise)}
+    />
+
```

Comment:
looks a lot like `BindStoreMixin` which can be customized to listen to custom events